src/skdh/context/motion_detect.py

```diff
@@ -38,42 +58,301 @@ def __init__(self, filter_cutoff=6, cov_threshold=0.022558169349678834):
+        High pass filter order. Data filtered prior to CoV calculation.
+    cov_threshold : float
+        Threshold computed rolling CoV to determine motion present.
+    sd_threshold : float
+        Threshold for max standard deviation to determine motion present.
+
+    """
+
+    def __init__(
+        self,
+        lp_filter_cutoff=3.0,
+        lp_filter_order=6,
+        hp_filter_cutoff=0.25,
+        hp_filter_order=1,
+        cov_threshold=0.41,
+        sd_threshold=0.013,
+    ):
+        super().__init__(
+            lp_filter_cutoff=lp_filter_cutoff,
+            lp_filter_order=lp_filter_order,
+            hp_filter_cutoff=hp_filter_cutoff,
+            hp_filter_order=hp_filter_order,
+            cov_threshold=cov_threshold,
+            sd_threshold=sd_threshold,
+        )
+        self.lp_filter_cutoff = lp_filter_cutoff
+        self.lp_filter_order = lp_filter_order
+        self.hp_filter_cutoff = hp_filter_cutoff
+        self.hp_filter_order = hp_filter_order
+        self.cov_threshold = cov_threshold
+        self.sd_threshold = sd_threshold
+
+    @handle_process_returns(results_to_kwargs=False)
+    def predict(self, time, accel, fs=None, **kwargs):
+        """
+        predict(time, accel, fs)
+
+        Function to detect periods of motion from an accelerometer signal.
+
+        Parameters
+        ----------
+        time : numpy.ndarray
+            (N, ) array of unix timestamps, in seconds.
+        accel : numpy.ndarray
+            (N, 3) array of acceleration, in units of 'g', collected at 20hz.
+        fs : float
+            Sampling rate. Default None. If not provided, will be inferred.
+
+        Returns
+        -------
+        results : dict
+            Results dictionary with motion predictions (1hz boolean; true indicates motion) and associated timestamps.
+
+        """
+        # check input requirements are met
+        time, accel, fs = self._check_input(time, accel, fs)
+
+        # Branch 1: Rolling CoV thresholding
+        # 1. Vector Magnitude
+        vmag = np.linalg.norm(accel, axis=1)
+
+        # 2. Low-pass filter the accelerometer vector magnitude signal to remove high frequency components
+        sos_lp = butter(
+            N=self.lp_filter_order,
+            Wn=2 * self.lp_filter_cutoff / fs,
+            btype="lowpass",
+            output="sos",
+        )
+        vmag_lp = sosfiltfilt(sos_lp, vmag, axis=0)
+
+        # 3. High-pass filter the accelerometer vector magnitude signal to remove DC component
+        sos_hp = butter(
+            N=self.hp_filter_order,
+            Wn=2 * self.hp_filter_cutoff / fs,
+            btype="highpass",
+            output="sos",
+        )
+        vmag_lp_hp = sosfiltfilt(sos_hp, vmag_lp, axis=0)
+
+        # 4. Calculate the 1s rolling coefficient of variation
+        rolling_std, rolling_mean = moving_sd(a=vmag_lp_hp, w_len=int(fs), skip=1)
+        rolling_cov = rolling_std / (np.abs(rolling_mean) + 0.01)
```

Comment:
0.01 from paper? since it differs from previous method 1e-12

Reply:
0.01 is from paper